### ISSUE TITLE
Add setup option to allow invalid certificates

### ISSUE THREAD

**Issue by phillipj** [phillipj/gitlab-search#3]
While having an on-premise hosted GitLab server, some organisations out in the wild might be running self signed not-verifiable TLS certificates for whatever reason. Currently this tools explodes badly when trying to communicate with such a server.

Being able to explicitly configure "ignore TLS certificate" upon setting up this tool, would avoid that mentioned explosion.

**Comment by phillipj** [comment]
Found that this isn't supported out of the box with the type defintions of bs-node-fetch. In fact, I couldn't find any bucklescript bindings that allows passing options down to Node.js's `https.request()` to ignore certificates.

Seeing that bs-axios has type definitions written in ReasonML, rather than Bucklescript as bs-node-fetch, I've decided to give bs-axios a try (hence https://github.com/phillipj/gitlab-search/commit/21ed8f925689665b6fb50cdf84f1cceba91b5954) and open a upstream PR to allows the needed configuration to be provided.

**Comment by phillipj** [comment]
For ref, what seems to be needed when using (bs-)axios seen from from JavaScript's point of view:

```js
const agent = new https.Agent({  
  rejectUnauthorized: false
});
axios.get('https://something.com/foo', { httpsAgent: agent });
```

https://github.com/axios/axios/issues/535#issuecomment-260841069

**Comment by phillipj** [comment]
Fixed and landed to master in 00dd76923ea1be962b4bc4bd9f345c638ce3975f
